test/functional/forknotify.py

```diff
@@ -41,11 +42,17 @@ def run_test(self):
         self.nodes[1].generate(1)
         self.sync_all()
 
-        with open(self.alert_filename, 'r', encoding='utf8') as f:
-            alert_text = f.read()
+        # Give bitcoind 10 seconds to write the alert notification
+        timeout = 10
+        while True:
```

Comment:
Maybe a bit cleaner to do something like (a couple of changes, feel free to pick and choose -- I was torn on the os.path.exists+getsize v.s. continual reading with open).

```python
        for t in xrange(100):
            if os.path.exists(self.alert_filename) and os.path.getsize(self.alert_filename):
                break
            if t != 99: time.sleep(0.1) 
        else:
            raise AssertionError("-alertnotify did not warn of up-version blocks")
        with open(self.alert_filename, 'r', encoding='utf8') as f:
            alert_text = f.read()
```